crypto/dh/dh_group_params.c

```diff
@@ -76,7 +75,6 @@ void ossl_dh_cache_named_group(DH *dh)
             dh->params.q = (BIGNUM *)ossl_ffc_named_group_get_q(group);
         /* cache the nid */
         dh->params.nid = ossl_ffc_named_group_get_uid(group);
-        dh->length = BN_num_bits(dh->params.q);
```

Comment:
What about keeping it with `if (dh->length == 0)`?

Reply:
There is no point. The gen code will use BN_num_bits(dh->params.q) if the length is not set. This is the main reason for setting it (i.e. to set it to something different from the default).